msdsn/conn_str.go

```diff
@@ -74,21 +78,40 @@ type Config struct {
 	PacketSize  uint16
 }
 
-func SetupTLS(certificate string, insecureSkipVerify bool, hostInCertificate string) (*tls.Config, error) {
+func SetupTLS(certificate string, insecureSkipVerify bool, hostInCertificate string, allowNonstandardHostname bool) (*tls.Config, error) {
 	var config tls.Config
+	config.ServerName = hostInCertificate
 	if certificate != "" {
 		pem, err := ioutil.ReadFile(certificate)
 		if err != nil {
 			return nil, fmt.Errorf("cannot read certificate %q: %v", certificate, err)
 		}
-		certs := x509.NewCertPool()
-		certs.AppendCertsFromPEM(pem)
-		config.RootCAs = certs
-	}
-	if insecureSkipVerify {
-		config.InsecureSkipVerify = true
+		if !allowNonstandardHostname {
```

Comment:
The core issue is that Go TLS does not consider a hostname with ":" to be valid. I generally agree, but here we are.

In this case, remove all code and documentation regarding the new option. rather, do a check in `config.ServerName` if it contains a ":". If that is the case, then do the alternative check.